### PR TITLE
Detecting file format using signature instead of file extension

### PR DESCRIPTION
**Pull request by Ark-kun**
Currently the backend detects the pipeline payload type by looking at the file extension.
If we merge the https://github.com/kubeflow/pipelines/pull/855 PR, the DSL compiler will start produce files in ZIP format, but unless the pipeline author changes the code (which they realistically won't do), the file name and extension will remain the same: `.tar.gz`. This will cause the backend to stop accepting those compiled pipelines which will frustrate our users.

This PR replaces format checks based on file name with checks based on file signatures. This fixes the backwards compatibility problem.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/919)

### CODE REVIEW

File: backend/src/apiserver/server/util.go [hunk 1]
```diff
@@ -54,20 +54,20 @@ func loadFile(fileReader io.Reader, maxFileLength int) ([]byte, error) {
...
 }
 
 func isYamlFile(fileName string) bool {
 	return strings.HasSuffix(fileName, ".yaml") || strings.HasSuffix(fileName, ".yml")
 }
 
-func isZipFile(fileName string) bool {
-	return strings.HasSuffix(fileName, ".zip")
+func isZipFile(compressedFile []byte) bool {
+	return len(compressedFile) > 2 && compressedFile[0] == '\x50' && compressedFile[1] == '\x4B' //Signature of zip file is "PK"
```
Comment (IronPan) [inline backend/src/apiserver/server/util.go]: could you add a test to see what would happen when a plain file starts with PK?

Comment (Ark-kun) [reply backend/src/apiserver/server/util.go]: I can, but I think it's already covered by the `TestDecompressPipelineZip_MalformattedZip` test. It's just a broken archive.

Comment (IronPan) [reply backend/src/apiserver/server/util.go]: the file is empty. could you create a file.txt with "PK" in the file?

Comment (Ark-kun) [reply backend/src/apiserver/server/util.go]: I've added another test. https://github.com/kubeflow/pipelines/pull/919/commits/a3c4715be3151bbbe7d61454101830709d2c874a

**Pull request merged**